src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -23,6 +23,7 @@ namespace Microsoft.AspNetCore.Mvc
     public class MvcOptions : IEnumerable<ICompatibilitySwitch>
     {
         private int _maxModelStateErrors = ModelStateDictionary.DefaultMaxAllowedErrors;
+        private int _maxValidationDepth = 512;
```

Comment:
Concerned this default limit will result in new errors for applications that are currently working just fine. Suggest either setting this to `int.MaxValue` or giving a special "unlimited" meaning to `-1` and using that as the default.

Otherwise, limit should be chosen based on a `CompatibilitySwitch` and be unlimited for 2.1 and earlier compat versions.

Reply:
Or `int?` 😆

Reply:
You nearly run in to insufficient stack space before you get to `int32.MaxValue`, but maybe `int?` is do-able.